src/instructlab/config/init.py

```diff
@@ -314,6 +314,19 @@ def walk_and_print_system_profiles(
     return cfg
 
 
+def is_hpu_available() -> bool:
+    # Third Party
+    import torch
+
+    # Check for HPU availability without errors
+    try:
+        hpu_available = hasattr(torch, "hpu") and torch.device("hpu") is not None
+        return hpu_available
+    # pylint: disable=broad-exception-caught
+    except Exception:
```

Comment:
Do we not get an expected Exception here?

Reply:
I do not believe so, looking in other parts other codebase, (vllm specifically), it seems we do this.